File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    # from collections import Counter
    # s = Counter(nums)
    
    dict = {}
    answer = 0
    
    for num in nums:
        if dict.get(num):
            dict[num] += 1
        else:
            dict[num] = 1   
    
    if len(dict) >= len(nums)/2:
        answer = len(nums)/2
    else:
        answer = len(dict)
        
    return answer
    

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    const n = nums.length / 2 ;
    const arr = []
    
    nums.forEach((value, index) => {
        if(index === 0) {
            arr.push(value)
        } else if(!arr.includes(value)) {
            arr.push(value)
        }
    })
    
    // 2, 2
    // 3, 3 
    // 3, 2
    
    return n > arr.length ? arr.length : n
}